src/client/pythonEnvironments/info/environmentInfoService.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { injectable } from 'inversify';
+import { EnvironmentType, PythonEnvironment } from '.';
+import { shellExecute } from '../common/externalDependencies';
+import { buildPythonExecInfo } from '../exec';
+import { getInterpreterInfo } from './interpreter';
+import { IWorkerPool, QueuePosition, WorkerPool } from './workerPool';
+
+export enum EnvironmentInfoServiceQueuePriority {
+    Default,
+    High
+}
+
+export const IEnvironmentInfoService = Symbol('IEnvironmentInfoService');
+export interface IEnvironmentInfoService {
+    getEnvironmentInfo(
+        interpreterPath: string,
+        priority?: EnvironmentInfoServiceQueuePriority
+    ): Promise<PythonEnvironment | undefined>;
+}
+
+@injectable()
+export class EnvironmentInfoService implements IEnvironmentInfoService {
+    private readonly cache: Map<string, PythonEnvironment>;
+    public constructor(private readonly workerPool?: IWorkerPool<string, PythonEnvironment | undefined>) {
+        this.cache = new Map<string, PythonEnvironment>();
+        if (!this.workerPool) {
+            this.workerPool = new WorkerPool<string, PythonEnvironment | undefined>(async (interpreterPath: string) => {
+                const interpreterInfo = await getInterpreterInfo(buildPythonExecInfo(interpreterPath), shellExecute);
+                if (interpreterInfo && interpreterInfo.version) {
+                    return {
+                        path: interpreterInfo.path,
+                        // Have to do this because the type returned by getInterpreterInfo is SemVer
+                        // But we expect this to be PythonVersion
+                        version: {
+                            raw: interpreterInfo.version.raw,
+                            major: interpreterInfo.version.major,
+                            minor: interpreterInfo.version.minor,
+                            patch: interpreterInfo.version.patch,
+                            build: interpreterInfo.version.build,
+                            prerelease: interpreterInfo.version.prerelease
+                        },
```

Comment:
Why does the `SevVer` type matter?  They are compatible.  If you are looking for a hard separation between `SemVer` and `PythonVersion` then I recommend using a helper function to make the separation stronger:

```suggestion
                        version: convertVersion(interpreterInfo.version),
```

Reply:
This was making testing harder, since SemVer was adding fields that were not there in PythonVersion.